presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1900,6 +1901,11 @@ public SystemSessionProperties(
                         "If both left and right side of join clause are varchar cast from int/bigint, remove the cast here",
                         featuresConfig.isRemoveRedundantCastToVarcharInJoin(),
                         false),
+                booleanProperty(
+                        REMOVE_MAP_CAST,
+                        "Remove map cast when possible",
+                        false,
+                        false),
```

Comment:
You might need to add this to features config to make this into a config property

Reply:
> You might need to add this to features config to make this into a config property

If this becomes a new feature property, please add documentation for it.

Reply:
@feilong-liu can you add a configuration property for this as per @jaystarshot's suggestion?  Session properties should always have a corresponding config property.

Reply:
> @feilong-liu can you add a configuration property for this as per @jaystarshot's suggestion? Session properties should always have a corresponding config property.

We are trying to move away from that model for optimizations that we know are safe (famous last words lol) to reduce noise. So feature config is only for features and not for optimizations.

Reply:
Adding a configuration doesn't seem especially noisy to me, but it does allow people to disable the feature for all queries if there ends up being a bug or regression in the optimizer (not everyone has infrastructure set up to easily tack session properties on to all user sessions if needed).